### PR TITLE
Fixed Windows-incompatible Bukkit world names for namespaced world keys.

### PR DESCRIPTION
**Pull request by VoChiDanh**
**Changelog**

- Fixed Windows-incompatible Bukkit world names for namespaced world keys.
- `Level#getName()` no longer returns the raw key string like `minecraft:void`.
- Added filesystem-safe Bukkit name mapping:
  - `minecraft:void` -> `minecraft_void`
  - `worlds:nested/void` -> `worlds_nested_void`
- Added `LevelNames` helper to centralize world-name sanitization.
- Added tests covering namespaced keys and keys containing `/`.
- Updated the 26.1.2 world loading path to override stale `LoadedWorldData.bukkitName()` values when old level data still contains an unsafe name.

**Principle**

Paper/Folia 26.1.2 changed world handling so the world key is now the real identity of a world. For example:

```text
world.key() = minecraft:void
```

That key is valid as a Minecraft namespaced key, and the plugin should keep using it for registry identity, dimensions, and internal world lookup.

The problem is that many Bukkit plugins, including WorldGuard, still use:

```java
world.getName()
```

as a legacy storage identifier. WorldGuard then creates paths like:

```text
plugins/WorldGuard/worlds/<world.getName()>/
```

On Windows, `:` is not allowed in folder names. So if `world.getName()` returns `minecraft:void`, WorldGuard crashes while trying to create:

```text
plugins/WorldGuard/worlds/minecraft:void/config.yml
```

The fix separates the two concepts:

```text
World identity/key: minecraft:void
Legacy Bukkit name: minecraft_void
```

So plugins that understand modern world keys can still use `world.key()`, while older plugins using `world.getName()` receive a filesystem-safe name.

WorldGuard will now store its data under:

```text
plugins/WorldGuard/worlds/minecraft_void/
```

instead of the invalid Windows path:

```text
plugins/WorldGuard/worlds/minecraft:void/
```

This preserves compatibility without changing the actual world key or the Paper 26.1.2 folder mechanism.

I tried on Folia 26.1.2
image
image

### CODE REVIEW

File: version-specifics/v26.1.2/src/main/java/net/thenextlvl/worlds/versions/v26_1_2/SimpleVersionHandler.java [hunk 1]
```diff
@@ -248,6 +248,14 @@ public CompletableFuture<World> createAsync(final Level level) {
...
                 name
         );
+        if (!loadedWorldData.bukkitName().equals(name)) {
+            loadedWorldData = new PaperWorldLoader.LoadedWorldData(
+                    name,
+                    loadedWorldData.uuid(),
+                    loadedWorldData.pdc(),
+                    loadedWorldData.levelOverrides()
+            );
+        }
```
Comment (NonSwag) [inline version-specifics/v26.1.2/src/main/java/net/thenextlvl/worlds/versions/v26_1_2/SimpleVersionHandler.java]: What is that supposed to do?
The legacy name is passed in L235 https://github.com/TheNextLvl-net/worlds/blob/353b9fae173f55f7b032f774b0877ba9f8477871/version-specifics/v26.1.2/src/main/java/net/thenextlvl/worlds/versions/v26_1_2/SimpleVersionHandler.java#L231-L238
The name used for registration doesn't have to match with the migration name

File: api/src/test/java/net/thenextlvl/worlds/LevelTest.java
Comment (NonSwag) [inline api/src/test/java/net/thenextlvl/worlds/LevelTest.java:1]: Kind of redundant micro test that isn't 100% reliable, the goal is to produce a valid path for Windows but CI currently only runs on Ubuntu.
We could change that, so it also runs on Windows and write a real test for it, but since this is such an edge case that can only be caused by the legacy name, I don't see it being useful after this is fixed once.
I would say, just remove this test.

File: api/src/main/java/net/thenextlvl/worlds/LevelNames.java
Comment (NonSwag) [inline api/src/main/java/net/thenextlvl/worlds/LevelNames.java:1]: This method is only invoked in one place ever so having a utility class/method for it is kinda redundant, just move it to the `Level#getName` method.

**Review by NonSwag — CHANGES_REQUESTED** [review]
Thanks for the PR

File: version-specifics/v26.1.2/src/main/java/net/thenextlvl/worlds/versions/v26_1_2/SimpleVersionHandler.java (hunk 1 above)
Comment (VoChiDanh) [reply version-specifics/v26.1.2/src/main/java/net/thenextlvl/worlds/versions/v26_1_2/SimpleVersionHandler.java]: WorldGuard saves by World#getName(), not by world.key().

If loadWorldData returns the old bukkitName `minecraft:void` after restart, WorldGuard looks for another folder and the flags seem lost. On Windows that name is also invalid as a path.

So the LoadedWorldData override is only to keep World#getName() stable as `minecraft_void`. It does not change the world key or the migration name.

File: api/src/test/java/net/thenextlvl/worlds/LevelTest.java
Comment (VoChiDanh) [reply api/src/test/java/net/thenextlvl/worlds/LevelTest.java:1]: I deleted it, thanks

File: api/src/main/java/net/thenextlvl/worlds/LevelNames.java
Comment (VoChiDanh) [reply api/src/main/java/net/thenextlvl/worlds/LevelNames.java:1]: I deleted it, thanks

File: version-specifics/v26.1.2/src/main/java/net/thenextlvl/worlds/versions/v26_1_2/SimpleVersionHandler.java (hunk 1 above)
Comment (NonSwag) [reply version-specifics/v26.1.2/src/main/java/net/thenextlvl/worlds/versions/v26_1_2/SimpleVersionHandler.java]: the legacy name and new name will most likely never match
also you are setting the name to the name so it doesn't do anything actually
the way i see it that change can be removed completely as the registered name is derived from `Level#getName` which is safe to use as a path on Windows

**Comment by NonSwag** [comment]
Thank you for the contribution :heart:
